pkg/reboot/calculator.go

```diff
@@ -76,12 +78,14 @@ func NewManagerSafeTimeCalculator(k8sClient client.Client) SafeTimeCalculator {
 }
 
 func (s *safeTimeCalculator) GetTimeToAssumeNodeRebooted() time.Duration {
+	minTime := s.minTimeToAssumeNodeRebooted
```

Comment:
In a nutshell we always return the greater between `timeToAssumeNodeRebooted` and `minTimeToAssumeNodeRebooted`.
In case it's not an agent we need to get `minTimeToAssumeNodeRebooted` from the config.

I can add a comment explaining why `minTimeToAssumeNodeRebooted` is fetched from config for the **non agent** use case.
Would that work or did you have anything else in mind ?

Reply:
I see it this way

```go
func (s *safeTimeCalculator) GetTimeToAssumeNodeRebooted() time.Duration {
	if !s.isAgent {
		minTime, _ = s.getMinSafeTimeFromConfig()
		if s.timeToAssumeNodeRebooted < minTime {
			return minTime
		}
	}

	return s.timeToAssumeNodeRebooted
}
```

other than that, I still believe `go` will benefit a lot from a ternary operator :grin:

Reply:
In that case for an agent use case where timeToAssumeNodeRebooted < minTime we will return a wrong value (i.e we will return timeToAssumeNodeRebooted instead of minTime)

Reply:
oh, right, I've been confused by `minTimeTo/timeTo`
sorry for the noise

Reply:
It's complicated and hard to understand, just took me some significant time to understand when which value is set or get or calculated...
While we get the config CR anyway now for the manager here, I guess we can at least remove the `SetTimeToAssumeNodeRebooted` method? IIUC, it's only used by the manager when reconciling the config... 🤔

Reply:
>It's complicated and hard to understand, just took me some significant time to understand when which value is set or get or calculated...

This is a good point, I think for me the first flag to indicate this was Carlo's question.
I'll have a crack at improving this with some comments/refactoring

>I guess we can at least remove the SetTimeToAssumeNodeRebooted method? IIUC, it's only used by the manager when reconciling the config... 🤔

Good idea, it should help simplifying the code